File: TeamCode/src/main/java/net/gearmaniacs/teamcode/motors/Intake.java
package net.gearmaniacs.teamcode.motors;

import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.DcMotorEx;
import com.qualcomm.robotcore.hardware.HardwareMap;

import net.gearmaniacs.teamcode.utils.IHardware;
import org.jetbrains.annotations.NotNull;

public final class Intake implements IHardware {

    private DcMotorEx left;
    private DcMotorEx right;

    @Override
    public void init(@NotNull HardwareMap hardwareMap) {
        HardwareMap.DeviceMapping<DcMotor> dcMotors = hardwareMap.dcMotor;
        left = (DcMotorEx) dcMotors.get("intake_left");
        right = (DcMotorEx) dcMotors.get("intake_right");
    }

    @Override
    public void start() {
    }

    @Override
    public void stop() {
    }

    @NotNull
    public final DcMotorEx getLeft() {
        return left;
    }

    @NotNull
    public final DcMotorEx getRight() {
        return right;
    }

    public final void setPowerAll(double power) {
        left.setPower(power);
        right.setPower(power);
    }
}

File: TeamCode/src/main/java/net/gearmaniacs/teamcode/hardware/motors/Intake.kt
package net.gearmaniacs.teamcode.hardware.motors

import com.qualcomm.robotcore.hardware.DcMotor
import com.qualcomm.robotcore.hardware.DcMotorEx
import com.qualcomm.robotcore.hardware.DcMotorSimple
import com.qualcomm.robotcore.hardware.HardwareMap
import net.gearmaniacs.teamcode.utils.IHardware
import net.gearmaniacs.teamcode.utils.getDevice

class Intake : IHardware {

    lateinit var left: DcMotorEx
        private set
    lateinit var right: DcMotorEx
        private set

    override fun init(hardwareMap: HardwareMap) {
        left = hardwareMap.getDevice("intake_left")
        right = hardwareMap.getDevice("intake_right")

        left.zeroPowerBehavior = DcMotor.ZeroPowerBehavior.BRAKE
        right.zeroPowerBehavior = DcMotor.ZeroPowerBehavior.BRAKE
    }

    fun setPowerAll(power: Double) {
        left.power = power
        right.power = -power
    }
}
